### ISSUE TITLE
fix(ml-spec): optimizePermittedRoles not applied to condition-specific overrides

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3724]
## Describe the bug

`getVersionResolvedARIA()` applies `optimizePermittedRoles()` to the **base** spec's `permittedRoles` (adding `none`↔`presentation` synonyms and ARIA 1.3 `image`↔`img` synonyms), but when a condition overrides `permittedRoles` in `getARIA()`, the overridden value **does not** go through `optimizePermittedRoles()`.

## Root cause

In `packages/@markuplint/ml-spec/src/algorithm/aria/get-aria.ts`:

```typescript
// Line 89-93: optimization applied to BASE spec only
function getVersionResolvedARIA(...) {
  aria = resolveVersion(spec, version);
  if (aria.permittedRoles !== false) {
    aria = {
      ...aria,
      permittedRoles: optimizePermittedRoles(aria.permittedRoles, version),
    };
  }
  // ...
}

// Line 63: condition override — NO optimization
for (const cond of conditionKeys) {
  if (matched) {
    permittedRoles = condARIA.permittedRoles ?? permittedRoles;
    // ↑ raw value from spec data, not optimized
  }
}
```

## Impact

- If a condition specifies `permittedRoles: ["presentation"]` but not `"none"`, writing `role="none"` would be incorrectly rejected (false positive).
- In ARIA 1.3, if a condition specifies `permittedRoles: ["img"]` but not `"image"`, writing `role="image"` would be incorrectly rejected.

Currently no known false positives because existing condition-specific `permittedRoles` arrays don't include `presentation`/`none`/`img`/`image`. However, this is a latent bug that will surface if such roles are added to condition-specific specs in the future.

## Fix

Apply `optimizePermittedRoles()` to condition-specific `permittedRoles` arrays in `getARIA()`, or move the optimization to `getPermittedRoles()` where the final list is assembled.

## Affected package

`@markuplint/ml-spec` — `get-aria.ts`

## Found during

Investigation of #3588 (input element permitted roles false positives).
